packages/amplify-cli/src/utils/team-provider-migrate.ts

```diff
@@ -0,0 +1,52 @@
+import { Context } from '../domain/context';
+import { stateManager, pathManager, PathConstants } from 'amplify-cli-core';
+import _ from 'lodash';
+import { externalAuthEnable } from 'amplify-category-auth';
+import { isYesFlagSet } from './headless-input-utils';
+import chalk from 'chalk';
+import { moveSecretsFromTeamProviderToDeployment } from './move-secrets-to-deployment';
+
+const message = `Amplify has been upgraded to handle secrets more securely by migrating some values in ${chalk.red(
+  PathConstants.TeamProviderInfoFileName,
+)} to ${chalk.green(PathConstants.DeploymentSecretsFileName)}
+You can create a back up of the ${chalk.red(PathConstants.TeamProviderInfoFileName)} file before proceeding.`;
+const hostedUIProviderCredsField = 'hostedUIProviderCreds';
+
+export const migrateTeamProviderInfo = async (context: Context): Promise<boolean> => {
+  // check if command executed in proj root and team provider has secrets
+  if (!isPulling(context) && pathManager.findProjectRoot() && teamProviderInfoHasAuthSecrets()) {
```

Comment:
it looks like it should be fine because this condition checks for `!isPulling()` but can you double check that this headless pull command still works with your changes: https://github.com/aws-amplify/amplify-cli/issues/3642#issuecomment-655804757

Reply:
I did a headless pull and init

Reply:
a headless pull with the social secrets specified in --categories?

Reply:
Yes

Reply:
there is a tagged release if you would like to make sure `@teamprv2`